woocommerce/woocommerce-ios
# WC Tracker: Add telemetry action

## Description

This PR adds telemetry endpoint support to report mobile app data (platform and version) to the Store.

## How

For telemetry endpoint usual Yosemite and Networking layers added. To reduce load on server:
- Action is dispatched only when wc-telemetry namespace available (we can check it in `RequirementsChecker`), WC 5.9.0 and up. This also prevents unwanted 404 errors in logs.
- Action is dispatched only once in 24h. Timeout is saved independently for every user’s store.

Server doesn't return any data in response in case of success. Jetpack wraps it in:
```
{
  "data": null
}
```
To ignore any kind of mapping here special `IgnoringResponseMapper` added.

Internal ref: p91TBi-6n5-p2

## Testing

iOS App:
1. Run it on WC <5.9.0. There shouldn't be any telemetry requests, including 404.
2. Install 5.9.0-rc.1 ([WooCommerce Beta Tester](https://wordpress.org/plugins/woocommerce-beta-tester/) plugin can help)
3. Switch on store with WC 5.9.0 and make sure only 1 telemetry request appeared in logs - for multiple app switches (background - foreground).
4. [Optional] Switch on different store with WC 5.9.0 and see same 1 telemetry request for that store.

[Optional] Verify on site:
1. Connect with ssh to the site after sending telemetry. 
2. Via WP-CLI run `wp eval 'var_dump(WC_Tracker::get_tracking_data());'`.
3. There should be a section like this:
```
["wc_mobile_usage"]=>
  array(1) {
    ["ios"]=>
    array(3) {
      ["platform"]=>
      string(3) "ios"
      ["version"]=>
      string(5) "7.8"
      ["last_used"]=>
      string(25) "2021-10-28T13:05:24+00:00"
    }
  }
```

---
Update release notes:
